src/daemon/plugin.c

```diff
@@ -651,7 +651,31 @@ static void *plugin_read_thread (void __attribute__((unused)) *args)
 	return ((void *) 0);
 } /* void *plugin_read_thread */
 
-static void start_read_threads (int num)
+static void set_thread_name(pthread_t tid, char const *name) {
+#if defined(HAVE_PTHREAD_SETNAME_NP) || defined(HAVE_PTHREAD_SET_NAME_NP)
+#if !defined(HAVE_PTHREAD_SETNAME_NP)
+# define pthread_setname_np pthread_set_name_np
+#endif
+
+	/* glibc limits the length of the name and fails if the passed string
+	 * is too long, so we truncate it here. */
+	char n[THREAD_NAME_MAX];
```

Comment:
Meh, I've moved the code around a bit and it was left in the wrong place. Fixed.